### PR TITLE
Make `align 16` work usefully

### LINKED ISSUE #1285: `ALIGN 16` inconsistently works or breaks
```asm
; test.asm
SECTION "test", ROM0;, ALIGN[16, 3]
    ;align 16, 3
    db 1,2,3
```
```asm
; test.link
ROM0
    org 3
    ;align 16, 3
    "test"
```
```bash
rgbasm -o test.o test.asm
rgblink -x -l test.link -o test.gb test.o
xxd -p test.gb
```

The above should all work and print `000000010203` with the `align`s uncommented. Instead:

- Uncommenting `SECTION ALIGN[16, 3]` causes this error when linking:
  ```
  The linker script assigns section "test" to address $0003, but it was already at $0000
  ```
- Uncommenting test.asm's `align 16, 3` works, but when `SECTION ALIGN[16, 3]` is also uncommented, it fails:
  ```
  Section's fixed address fails required alignment (PC = $0000)
  ```
- ~~Uncommenting test.link's `align 16, 3` *hangs indefinitely*! 😢~~ EDIT: fixed by #1292. No errors when uncommented.

So, `SECTION ALIGN[16, N]` should set the address to *N*, not 0; and linker script `align 16, N` should work at all.

### PR DESCRIPTION
**Pull request by ISSOtm**
Fixes #1285.

### CODE REVIEW

File: src/asm/section.cpp [hunk 1]
```diff
@@ -355,7 +355,7 @@ static struct Section *getSection(char const *name, enum SectionType type, uint3
 		} else if (alignment == 16) {
 			// Treat an alignment of 16 as being fixed at address 0
 			alignment = 0;
-			org = 0;
+			org = alignOffset;
```
Comment (Rangi42) [inline src/asm/section.cpp:358]: The comment above this needs updating to match.

File: src/asm/section.cpp [hunk 2]
```diff
@@ -508,6 +508,14 @@ void sect_AlignPC(uint8_t alignment, uint16_t offset)
 					- offset) % alignSize) {
 		error("Section's alignment fails required alignment (offset from section start = $%04"
 			PRIx32 ")\n", curOffset);
+	} else if (alignment >= 16) {
+		// Treat an alignment large enough as fixing the address.
+		// Note that this also ensures that a section's alignment never becomes 16 or greater.
+		if (alignment > 16) {
+			error("Alignment must be between 0 and 16, not %u\n", alignment);
```
Comment (Rangi42) [inline src/asm/section.cpp:515]: `getSection` caps the alignment at 16 when it does this error; so probably this should too. Then a test case `align 17, 42` would error (so fail to assemble) but continue non-fatally like `align 16, 42`.
```suggestion
			error("Alignment must be between 0 and 16, not %u\n", alignment);
			alignment = 16;
```

**Review by Rangi42 — CHANGES_REQUESTED** [review]

File: src/asm/section.cpp (hunk 2 above)
Comment (ISSOtm) [reply src/asm/section.cpp:515]: That would be a dead write, since `alignment` is not used afterwards.

Comment (Rangi42) [reply src/asm/section.cpp:515]: Okay, I see now; it already treats alignment >16 that way.

**Review by Rangi42 — APPROVED** [review]
Looks good. :) Thank you!

**Pull request merged**